worlds/smz3/__init__.py

```diff
@@ -401,7 +420,7 @@ def FrontFillItemInOwnWorld(self, itemPool, itemType):
             raise Exception(f"Tried to front fill {item.Name} in, but no location was available")
         
         location.Item = item
-        itemFromPool = next((i for i in self.world.itempool if i.player == self.player and i.name == item.Type.name), None)
+        itemFromPool = next((i for i in self.world.itempool if i.player == self.player and i.name == item.Type.name and i.advancement == item.Progression), None)
         if itemFromPool is not None:
             self.world.get_location(location.Name, self.player).place_locked_item(itemFromPool)
             self.world.itempool.remove(itemFromPool)
```

Comment:
Its called from pre_fill. What you say is true but since its a port, I prefer keeping the code as close as possible as the original to facilitate updates.